gobblin-runtime/src/main/java/gobblin/runtime/MultiConverter.java

```diff
@@ -80,82 +78,111 @@ public Object convertSchema(Object inputSchema, WorkUnitState workUnit)
   }
 
   /**
-   * A type of {@link java.util.Iterator} to be used with {@link MultiConverter}.
+   * A type of {@link java.util.Iterator} to be used with {@link MultiConverter}. This method works by maintaining a
```

Comment:
Yes, I agree. That is a much simpler solution. I just finished the new implementation and it passes all the tests. I will update the pull request.